### ISSUE TITLE
Code does not follow the Moodle coding style

### ISSUE THREAD

**Issue by mikemcgowan** [davidpesce/moodle-logstore_xapi#117]
Is there any particular reason why the code style doesn't follow the [Moodle](https://docs.moodle.org/dev/Coding_style) [coding](https://docs.moodle.org/dev/CodeSniffer) [style](https://github.com/moodlehq/moodle-local_codechecker)?

Infact, recent work in v2.0.0 has moved further away from the Moodle `snake_case` style to more like the PSR2 `camelCase` style (for variable names).

Since this is a Moodle plugin, ideally, the Moodle coding style should be adhered to ...

**Comment by ryasmi** [comment]
Just because PSR2 is the default for things like code climate and the Emitter was designed to be used outside of Moodle.

**Comment by mikemcgowan** [comment]
Should PSR2 be the style for everything then? And if so, is it worth putting that somewhere like the main README.md, or the dev documentation or something?

Because the code doesn't lint to PSR2 either, e.g. these two come up everywhere:

```
  3 | ERROR | [x] Opening brace of a class must be on the line after the definition
  4 | ERROR | [x] Opening brace should be on a new line
```

**Comment by ryasmi** [comment]
Best thing to do now I think, would be to conform to Moodle style guide if:
1. We have to conform in order to be on the plugin directory.
2. We can configure Code Climate to use the Moodle style guide.

If those two things are true, then we should document that we are aiming to conform to that style guide and note that Code Climate will enforce that style guide on pull requests and commits.

If the second is true, but the first of those two things is not true, then we should use PSR2 or some other standard that Code Climate can enforce.

If the first is true, but the second of those two things is not true, then we should find some other way to enforce a style guide and drop Code Climate.

**Comment by mikemcgowan** [comment]
[According to this](https://docs.codeclimate.com/v1.0/docs/phpcodesniffer), Code Climate does support PHP Code Sniffer, but doesn't support the Moodle standard.

Moodlerooms have [a plugin](https://github.com/moodlerooms/moodle-plugin-ci) which allows linting Moodle plugin code against the Moodle coding style using Travis. So maybe that could be used instead of Code Climate?

(Whichever decision you take, I'm happy to submit a PR that fixes the codebase to agreed standard, whether it's Moodle or PSR2.)

**Comment by ryasmi** [comment]
Awesome, if you can make a PR that uses the Moodle rooms plugin, myself and @davidpesce can probably review that.

**Comment by garemoko** [comment]
I've set up the Moodle coding style using Travis on the xapi launch plugin (and the Watershed Moodle plugin) and my preference would be to go for that style. 

Side note: the xapi launch plugin is currently failing because the Brazil Portuguese translation is currently failing due to having "duplicated code" from the Portugal Portuguese translation. I haven't had time to dig into whether that's a problem with the moodle code checker or whether the Brazil Portuguese translation should somehow reference the Portugal Portuguese translation.

**Comment by garemoko** [comment]
If it helps, here's the relevant file: https://github.com/garemoko/moodle-mod_tincanlaunch/blob/master/.travis.yml

**Comment by mikemcgowan** [comment]
Fixed by #119
